RDKShell/RDKShell.cpp

```diff
@@ -606,6 +609,10 @@ namespace WPEFramework {
                mLaunchEnabled = false;
             }
 
+            suspendedOrHibernatedAppsMutex.lock();
```

Comment:
Might be good to use the #ifdef HIBERNATE_SUPPORT_ENABLED everywhere we use suspendOrHibernateApps array and mutex.